File: swift/1st_ex/main.swift
import Foundation

func err(_ exc: String) -> Never {
    fatalError(exc)
}

func initVal() -> Int {
    guard let line = readLine(), let val = Int(line), val > 0 else {
        err("Invalid input")
    }
    return val
}

func funcCalc(S: Int, N: Int, M: Int) -> Int {
    if S * M > (S - S / 7) * N {
        return -1
    }
    return Int(ceil(Double(S * M) / Double(N)))
}

let S = initVal()
let N = initVal()
let M = initVal()

print(funcCalc(S: S, N: N, M: M))

File: go/1st_ex/main.go
package main

import (
	"fmt"
	"math"
)

func err(exc string) {
	panic(exc)
}

func initVal() int {
	var val int
	_, err := fmt.Scanf("%d", &val)
	if err != nil || val <= 0 {
		panic("Invalid input") 
	}
	return val
}

func funcCalc(S, N, M int) int {
	if S*M > (S - S/7)*N {
		return -1
	}
	return int(math.Ceil(float64(S*M) / float64(N)))
}

func main() {
	S := initVal()
	N := initVal()
	M := initVal()

	fmt.Println(funcCalc(S, N, M))
}
